syntactic-paths/src/main/java/com/palantir/util/syntacticpath/Path.java

```diff
@@ -330,13 +322,52 @@ private String toStringInternal() {
             String prefix = isAbsolute ? SEPARATOR : "";
             String suffix = isFolder ? SEPARATOR : "";
 
-            return prefix + PATH_JOINER.join(segments) + suffix;
+            return prefix + join(segments) + suffix;
         }
     }
 
     /** Returns the string representation of this (non-normalized) path. */
     @Override
     public String toString() {
-        return stringRepresentation.get();
+        return stringRepresentation;
+    }
+
+    static String join(List<String> segments) {
```

Comment:
Curious as to why these should be more generic than necessary. In the guava version we had created a specialized splitting function by creating a `Splitter.on(CHAR)`